index.js

```diff
@@ -10,24 +11,21 @@ module.exports = (opts, types, defaults) => {
 	conf.add(Object.assign({}, opts), 'cli');
 	const warnings = [];
 
-	if (require.resolve.paths) {
-		const paths = require.resolve.paths('npm');
-		// Assume that last path in resolve paths is builtin modules directory
-		let npmPath;
-		try {
-			npmPath = require.resolve('npm', {paths: paths.slice(-1)});
-		} catch (error) {
-			// Error will be thrown if module cannot be found.
-			// We should ignore that error.
-		}
+	let npmPath;
+	try {
+		// e.g. /usr/local/lib/node_modules
+		npmPath = execa.sync('npm', ['root', '-g']).stdout
```

Comment:
this would add about 1/3 of a second to every pnpm command. We can't solve the issue with this solution.

Reply:
How about moving it to catch? So it won't affect the normal cases

Reply:
I would rather print some warnings with instructions how to solve the issue in a proper way.

Reply:
Thanks.What's your suggestions about it? I'd like to update this pr to add some warnings. I would rather advise users to add a new npmrc in HOME to overwrite the built-in npmrc

Reply:
If it helps, then I am ok with that.

And alternative solution might be to find the right location without running `npm root -g`